app/controllers/api/user_books_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Api::UserBooksController < ApplicationController
+  before_action :authenticate_user!
+  before_action :get_user_book, only: [:destroy, :update, :show]
+
+  def index
+    books = params[:finish_date] == nil ? UserBook.to_read(current_user) : UserBook.finished(current_user)
```

Comment:
Please do not write business logic in the controller. This is a perfect opportunity to create a service. 

something like:

`app/services/user_books/query_service.rb`

Then the query_service should be intelligent enough to know what to return. So:

```
books = UserBooks::QueryService.new(params).search
```

Lastly, I don't see why you need to return an error if the list is empty. If the list is empty, then show nothing in the frontend.